### ISSUE TITLE
Provide ZAP Docker image for `windows/amd64`

### ISSUE THREAD

**Issue by Raulster24** [zaproxy/zaproxy#8468]
### Is your feature request related to a problem? Please describe.

No docker images for Windows engine, all distros images are linux

<img width="467" alt="image" src="https://github.com/zaproxy/zaproxy/assets/5059765/038e95da-45c3-4ac4-9287-0abde2d3fc00">

### Describe the solution you'd like

Create a docker image to be able to run on windowsWindows

### Describe alternatives you've considered

Need to run in azure devops pipeline and I have only option to run and its failing, on docker desktop I can easily change the windows to linux engine and do that. Look for any alternative solution but there is none

### Would you like to help fix this issue?

- [ ] Yes

**Comment by psiinon** [comment]
This is not currently on our roadmap, and so would be a good candidate for a [sponsored development](https://www.zaproxy.org/support/#sponsored-developments)

**Comment by kingthorin** [comment]
You should be able to leverage WSL.

https://stackoverflow.com/a/64759746/7718222

> It is now possible to run Linux docker images on Windows 10 natively using Windows Subsystem for Linux 2 (WSL2).
>
>Steps:
>
>    Install WSL2.
>    Install Docker for Windows.
>    Go to Docker Settings -> General and check mark "Use the WSL 2 based engine".
>
> Source: https://learn.microsoft.com/en-us/windows/wsl/tutorials/wsl-containers

**Comment by Raulster24** [comment]
@kingthorin can you please tell me more on this as in how to leverage WSL ?

**Comment by Raulster24** [comment]
@psiinon may be a very simple question, but can we run API scan through command line by installing ZAP proxy on my agent machine? Please suggest

**Comment by kingthorin** [comment]
> @kingthorin can you please tell me more on this as in how to leverage WSL? Any reference

Just click the little triangle beside the text and it'll expand a section with the details...

**Comment by psiinon** [comment]
> @psiinon may be a very simple question, but can we run API scan through command line by installing ZAP proxy on my agent machine? Please suggest

The API scan is limited to docker.
However you can do the same thing using the automation framework: https://www.zaproxy.org/docs/automate/automation-framework/ which can be run from the commandline and is much more flexible.

**Comment by Raulster24** [comment]
@psiinon Thanks, I downloaded the sh file to use it for the command line but I am getting this error when I execute it

![image](https://github.com/zaproxy/zaproxy/assets/5059765/7739f141-cb2e-4698-9e05-31cae580feef)

![image](https://github.com/zaproxy/zaproxy/assets/5059765/18df53b3-8324-462d-aeab-634e85f47f70)

**Comment by psiinon** [comment]
Thats a Linux shell script, and you appear to be trying to run it in Windows, rather than in WSL...

**Comment by Raulster24** [comment]
@psiinon  yes correct my bad, i used the WSL command now available in windows to switch to one of the linux distro, Thanks much.

**Comment by Raulster24** [comment]
@psiinon I looked into the link that you provided. I see that for openapi we have only importurl option, even after I import how can I see the list of URLs imported plus I also want to run the scan similar to API scan using docker image. Can you please direct me to the reference?

![image](https://github.com/zaproxy/zaproxy/assets/5059765/6fc8495d-4362-483a-a49d-4e2559bc4c5c)

**Comment by psiinon** [comment]
You are looking at the API docs.
I linked to the AF docs before: https://www.zaproxy.org/docs/automate/automation-framework/
They include an openapi job: https://www.zaproxy.org/docs/desktop/addons/openapi-support/automation/

Can you ask these sort of questions on the user group: https://groups.google.com/group/zaproxy-users
They do not appear to be directly relevant to the issue, ie providing a ZAP docker image for windows/amd64
